lib/galaxy/datatypes/binary.py

```diff
@@ -1528,6 +1532,35 @@ def display_peek(self, dataset):
         except Exception:
             return "HDF5 Model (%s)" % (nice_size(dataset.get_size()))
 
+    def display_data(self, trans, dataset, preview=False, filename=None, to_ext=None, **kwd):
+        preview = util.string_as_bool(preview)
+
+        if to_ext or not preview:
+            to_ext = to_ext or dataset.extension
+            return self._serve_raw(trans, dataset, to_ext, **kwd)
+
+        rval = {}
+        try:
+            with h5py.File(dataset.file_name, "r") as handle:
+                rval['Attributes'] = {}
+                for k in handle.attrs.keys():
+                    if k in ('-URL-', '-repr-'):
+                        continue
+                    v = util.unicodify(handle.attrs.get(k))
+                    rval['Attributes'][k] = v
+        except Exception as e:
+            log.warning(e)
+            pass
+
+        config = self.get_config_string(dataset.file_name)
+        rval['Config'] = json.loads(config)
```

Comment:
I think this should be conditional on the value of `config`. And `get_config_string` doesn't have to return `"{}"`: if you need it to be a str repr of a dict even if it's empty, you can do this:
```py
rval['Config'] = json.loads(config) if config else '{}'
```